### PR TITLE
fix(frontend/shared): fix Profile screen tm-42

### PR DESCRIPTION
**Pull request by Sytnikovakatya**
- Change validation schema according to updated specification
- Cancel button allows the user to undo all changes made
- The user is prevented from leaving the page without saving the changes — the "Please, update your changes before leaving” message appears as a toast-message
-  If the changes are successfully saved, the “Changes are saved.” message appears as a toast-message
-  Update styles of User foto on the header to repeat profile foto's styles
Screenshot_1
Screenshot_2
Screenshot_3

### CODE REVIEW

**Review by YourTeaChan — APPROVED** [review]

File: apps/frontend/src/pages/profile/profile.tsx [hunk 1]
```diff
@@ -24,14 +30,27 @@ const Profile: React.FC = () => {
...
-			firstName: user.firstName,
-			lastName: user.lastName,
-		},
+		defaultValues: formData,
 		validationSchema: userProfileValidationSchema,
 	});
 
+	const handleOnChange = useCallback(
+		(event: React.BaseSyntheticEvent): void => {
+			if (event.target instanceof HTMLInputElement) {
```
Comment (Oleksandra-Masiuk) [inline apps/frontend/src/pages/profile/profile.tsx:47]: Why do we need this `if`?

File: apps/frontend/src/pages/profile/profile.tsx [hunk 2]
```diff
@@ -64,17 +83,48 @@ const Profile: React.FC = () => {
 		[handleSubmit, handleInputChange],
 	);
 
+	const handleCancelChanges = useCallback(() => {
```
Comment (what1s1ove) [inline apps/frontend/src/pages/profile/profile.tsx:86]: can we just use a button with a type "reset" ?

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/profile/profile.tsx (hunk 2 above)
Comment (what1s1ove) [reply apps/frontend/src/pages/profile/profile.tsx:86]: or this use-form API
https://react-hook-form.com/docs/useform/reset

File: apps/frontend/src/libs/hooks/use-app-form/use-app-form.hook.ts [hunk 3]
```diff
@@ -23,6 +23,7 @@ type ReturnValue<T extends FieldValues = FieldValues> = {
 	control: Control<T, null>;
 	errors: FieldErrors<T>;
 	handleSubmit: UseFormHandleSubmit<T>;
+	reset: () => void;
```
Comment (fshabanov) [inline apps/frontend/src/libs/hooks/use-app-form/use-app-form.hook.ts:26]:
```suggestion
	reset: UseFormReset<T>;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/profile/profile.tsx (hunk 1 above)
Comment (Sytnikovakatya) [reply apps/frontend/src/pages/profile/profile.tsx]: Thanks, I've changed it

**Review by Vfivew — APPROVED** [review]

File: apps/frontend/src/pages/profile/profile.tsx [hunk 4]
```diff
@@ -64,17 +81,40 @@ const Profile: React.FC = () => {
 		[handleSubmit, handleInputChange],
 	);
 
+	const handleResetForm = useCallback(() => {
+		reset();
+	}, [reset]);
+
+	let blocker = useBlocker(
```
Comment (what1s1ove) [inline apps/frontend/src/pages/profile/profile.tsx:88]:
```suggestion
```
let's do not add waring functionality now 
good that you think about it, but not for now. i think you can remove this `useBlocker ` at all

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Oleksandra-Masiuk — APPROVED** [review]

**Review by GEOFARL — APPROVED** [review]

File: apps/frontend/src/pages/profile/profile.tsx [hunk 5]
```diff
@@ -74,7 +78,10 @@ const Profile: React.FC = () => {
 							<Image
 								alt="avatar"
 								className={styles["profile-image"]}
+								height="133"
```
Comment (fshabanov) [inline apps/frontend/src/pages/profile/profile.tsx:81]: Did you test this size in mobile view as well?

Comment (what1s1ove) [reply apps/frontend/src/pages/profile/profile.tsx:81]: it okay to add it in HTML, you just need to change the values in CSS as well

Comment (Sytnikovakatya) [reply apps/frontend/src/pages/profile/profile.tsx:81]: ok, I will change it

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
@Sytnikovakatya merge main branch into yours

**Pull request merged**